app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -521,8 +522,14 @@ def find_organization_from_product
     end
 
     def find_content
-      if params[:content_type]
-        @content = @repository.units_for_removal(params[:ids], params[:content_type])
+      content_type = params[:content_type]
+      repo_content_types = @repository.repository_type.content_types.collect { |type| type.label }
+      unless repo_content_types.include?(content_type)
+        fail HttpErrors::BadRequest, _("Content type %{content_type} is incompatible with repositories of type %{repo_type}" %
+                                       { content_type: content_type, repo_type: @repository.content_type })
+      end
```

Comment:
This logic looks like its appears twice, maybe we could move it to a function in api_controller?